src/ddprof_worker.cc

```diff
@@ -88,9 +88,26 @@ DDRes report_lost_events(DDProfContext &ctx) {
   return {};
 }
 
+inline int64_t get_mono_to_realtime_offset(
+    const std::chrono::steady_clock::time_point &steady_now) {
+  // Finds the offset between the two clocksources.  This doesn't account for
+  // NTP adjustments for now.
+  timespec ts_monotonic_raw{};
+
+  clock_gettime(CLOCK_MONOTONIC_RAW, &ts_monotonic_raw);
+
+  auto ns_realtime = steady_now.time_since_epoch().count();
+  auto ns_monotonic_raw = timespec_to_duration(ts_monotonic_raw).count();
+
+  return ns_realtime - ns_monotonic_raw;
+}
+
 inline void export_time_set(DDProfContext &ctx) {
-  ctx.worker_ctx.send_time =
-      std::chrono::steady_clock::now() + ctx.params.upload_period;
+  auto steady_now = std::chrono::steady_clock::now();
```

Comment:
Not sure it's a good idea to reuse `std::chrono::steady_clock::now()` value, just let `get_mono_to_realtime_offset` call again `std::chrono::steady_clock::now()` (it should be `std::chrono::system_time::now()` anyway), and cost does not matter since it's called once per minute.

Reply:
Agreed, fixed